llvm/lib/Target/RISCV/RISCVInstrInfoA.td

```diff
@@ -308,7 +308,65 @@ class PseudoMaskedAMOMinMaxPat<Intrinsic intrin, Pseudo AMOInst>
           (AMOInst GPR:$addr, GPR:$incr, GPR:$mask, GPR:$shiftamt,
            timm:$ordering)>;
 
-let Predicates = [HasStdExtA] in {
+let Predicates = [HasStdExtZalrsc, NoStdExtZaamo] in {
```

Comment:
There is a comment around line 161 above where `HasAtomicLdSt` is used that says `Atomic load/store are available under both +a and +force-atomics.`. This comment needs to be updated to +zalrsc